File: app/src/main/java/com/thinkzi/weather/ui/view/adapter/WeatherInformationUIModelListAdapter.java
package com.thinkzi.weather.ui.view.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.thinkzi.weather.ui.R;
import com.thinkzi.weather.ui.databinding.ItemWeatherInformationBinding;
import com.thinkzi.weather.ui.model.WeatherInformationUIModel;

import java.util.List;

public class WeatherInformationUIModelListAdapter extends RecyclerView.Adapter<WeatherInformationUIModelListAdapter.WeatherInformationUIModelViewHolder>{

    // application context
    private Context _context;

    private List<WeatherInformationUIModel> _weatherInformationUIModelList;

    public WeatherInformationUIModelListAdapter(Context _context, List<WeatherInformationUIModel> _weatherInformationUIModelList) {
        this._context = _context;
        this._weatherInformationUIModelList = _weatherInformationUIModelList;
    }

    @NonNull
    @Override
    public WeatherInformationUIModelViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return(new WeatherInformationUIModelViewHolder(ItemWeatherInformationBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false)));
    }

    @Override
    public void onBindViewHolder(@NonNull WeatherInformationUIModelViewHolder holder, int position) {
        holder.bind(_weatherInformationUIModelList.get(position));
    }

    @Override
    public int getItemCount() {
        return _weatherInformationUIModelList.size();
    }

    public void setWeatherInformationUIModelList(List<WeatherInformationUIModel> _weatherInformationUIModelList) {
        this._weatherInformationUIModelList = _weatherInformationUIModelList;
        notifyDataSetChanged();
    }

    class WeatherInformationUIModelViewHolder extends RecyclerView.ViewHolder {

        private final ItemWeatherInformationBinding _itemWeatherInformationBinding;

        WeatherInformationUIModelViewHolder(ItemWeatherInformationBinding _itemWeatherInformationBinding) {
            super(_itemWeatherInformationBinding.getRoot());
            this._itemWeatherInformationBinding = _itemWeatherInformationBinding;
        }

        void bind(final WeatherInformationUIModel _weatherInformationUIModel) {
            /*
            _itemWeatherInformationBinding.lblDateTime.setText(_context.getString(R.string.datetime) + " : " + _weatherInformationUIModel.getDateTime().toString("HH:mm:ss dd/MM/yyyy"));
            _itemWeatherInformationBinding.lblTemperature.setText(_context.getString(R.string.temperature) + " : " + String.valueOf(_weatherInformationUIModel.getTemperature().getTemperature()));
            _itemWeatherInformationBinding.lblMinTemperature.setText(_context.getString(R.string.min_temperature) + " : " + String.valueOf(_weatherInformationUIModel.getTemperature().getMinTemperature()));
            _itemWeatherInformationBinding.lblMaxTemperature.setText(_context.getString(R.string.max_temperature) + " : " + String.valueOf(_weatherInformationUIModel.getTemperature().getMaxTemperature()));
            _itemWeatherInformationBinding.lblCloudiness.setText(_context.getString(R.string.cloudiness) + " : " + String.valueOf(_weatherInformationUIModel.getCloud().getCloudiness()));
            _itemWeatherInformationBinding.lblHumidity.setText(_context.getString(R.string.humidity) + " : " + String.valueOf(_weatherInformationUIModel.getHumidity().getHumidity()));
            _itemWeatherInformationBinding.lblPressure.setText(_context.getString(R.string.pressure) + " : " + String.valueOf(_weatherInformationUIModel.getPressure().getPressure()));
            _itemWeatherInformationBinding.lblRainVolume.setText(_context.getString(R.string.rain_volume) + " : " + String.valueOf(_weatherInformationUIModel.getRain().getRainVolume()));
            _itemWeatherInformationBinding.lblSnowVolume.setText(_context.getString(R.string.snow_volume) + " : " + String.valueOf(_weatherInformationUIModel.getSnow().getSnowVolume()));
            _itemWeatherInformationBinding.lblWindSpeed.setText(_context.getString(R.string.wind_speed) + " : " + String.valueOf(_weatherInformationUIModel.getWind().getSpeed()));
            _itemWeatherInformationBinding.lblWindDirection.setText(_context.getString(R.string.wind_direction) + " : " + String.valueOf(_weatherInformationUIModel.getWind().getDirection()));
            */

            _itemWeatherInformationBinding.setWeatherInformationUIModel(_weatherInformationUIModel);

            _itemWeatherInformationBinding.executePendingBindings();

        }

    }
}

File: app/src/main/java/com/thinkzi/weather/ui/view/adapter/WeatherInformationUIModelListAdapter.kt
package com.thinkzi.weather.ui.view.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

import com.thinkzi.weather.ui.R
import com.thinkzi.weather.ui.databinding.ItemWeatherInformationBinding
import com.thinkzi.weather.ui.model.WeatherInformationUIModel

internal class WeatherInformationUIModelListAdapter(// application context
    private val _context: Context,
    private var _weatherInformationUIModelList: List<WeatherInformationUIModel>?
) : RecyclerView.Adapter<WeatherInformationUIModelListAdapter.WeatherInformationUIModelViewHolder>() {

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): WeatherInformationUIModelViewHolder {
        return WeatherInformationUIModelViewHolder(
            ItemWeatherInformationBinding.inflate(
                LayoutInflater.from(parent.context),
                parent,
                false
            )
        )
    }

    override fun onBindViewHolder(holder: WeatherInformationUIModelViewHolder, position: Int) {
        holder.bind(_weatherInformationUIModelList!![position])
    }

    override fun getItemCount(): Int {
        return _weatherInformationUIModelList!!.size
    }

    fun setWeatherInformationUIModelList(_weatherInformationUIModelList: List<WeatherInformationUIModel>) {
        this._weatherInformationUIModelList = _weatherInformationUIModelList
        notifyDataSetChanged()
    }

    internal inner class WeatherInformationUIModelViewHolder(private val _itemWeatherInformationBinding: ItemWeatherInformationBinding) :
        RecyclerView.ViewHolder(_itemWeatherInformationBinding.root) {

        fun bind(_weatherInformationUIModel: WeatherInformationUIModel) {
            _itemWeatherInformationBinding.lblDateTime.text =
                _context.getString(R.string.datetime) + " : " + _weatherInformationUIModel.dateTime.toString(
                    "HH:mm:ss dd/MM/yyyy"
                )
            _itemWeatherInformationBinding.lblTemperature.text =
                _context.getString(R.string.temperature) + " : " + _weatherInformationUIModel.temperature.temperature.toString()
            _itemWeatherInformationBinding.lblMinTemperature.text =
                _context.getString(R.string.min_temperature) + " : " + _weatherInformationUIModel.temperature.minTemperature.toString()
            _itemWeatherInformationBinding.lblMaxTemperature.text =
                _context.getString(R.string.max_temperature) + " : " + _weatherInformationUIModel.temperature.maxTemperature.toString()
            _itemWeatherInformationBinding.lblCloudiness.text =
                _context.getString(R.string.cloudiness) + " : " + _weatherInformationUIModel.cloud.cloudiness.toString()
            _itemWeatherInformationBinding.lblHumidity.text =
                _context.getString(R.string.humidity) + " : " + _weatherInformationUIModel.humidity.humidity.toString()
            _itemWeatherInformationBinding.lblPressure.text =
                _context.getString(R.string.pressure) + " : " + _weatherInformationUIModel.pressure.pressure.toString()
            _itemWeatherInformationBinding.lblRainVolume.text =
                _context.getString(R.string.rain_volume) + " : " + _weatherInformationUIModel.rain.rainVolume.toString()
            _itemWeatherInformationBinding.lblSnowVolume.text =
                _context.getString(R.string.snow_volume) + " : " + _weatherInformationUIModel.snow.snowVolume.toString()
            _itemWeatherInformationBinding.lblWindSpeed.text =
                _context.getString(R.string.wind_speed) + " : " + _weatherInformationUIModel.wind.speed.toString()
            _itemWeatherInformationBinding.lblWindDirection.text =
                _context.getString(R.string.wind_direction) + " : " + _weatherInformationUIModel.wind.direction.toString()

            _itemWeatherInformationBinding.executePendingBindings()

        }

    }
}
